transform.js

```diff
@@ -5,6 +5,9 @@ module.exports = transform => async function * (source) {
   // In a transform the sink and source are connected, an error in the sink
   // will be thrown in the source also. Catch the sink error to avoid unhandled
   // rejections and yield from the source.
-  duplex.sink(source).catch(_ => {})
+  let sinkError
+  duplex.sink(source).catch(err => { sinkError = err })
+
   yield * duplex.source
```

Comment:
I'm just wondering if we should actually throw as soon as we can when we get an error instead of when the source has completed? i.e. expand this out with a for await and check on every iteration?

Reply:
I don't think it makes a big difference, because the error does destroy the Transform stream, causing it's asyncIterator to quickly become done, so usually the very next call to generator.next() will throw the error.

There maybe edge cases where destroy doesn't actually make the stream stop producing data. I could write some tests to verify that, but I doubt it'll be a big issue.